tests/kafkatest/tests/connect/connect_rest_test.py

```diff
@@ -15,18 +15,47 @@
 
 from kafkatest.tests.kafka_test import KafkaTest
 from kafkatest.services.connect import ConnectDistributedService, ConnectRestError
-from kafkatest.utils.util import retry_on_exception
 from ducktape.utils.util import wait_until
 import subprocess
 import json
 import itertools
 
 
+def merge_two_dicts(x, y):
+    z = x.copy()
+    z.update(y)
+    return z
+
+
 class ConnectRestApiTest(KafkaTest):
     """
     Test of Kafka Connect's REST API endpoints.
     """
 
+    FILE_SOURCE_CONNECTOR = 'org.apache.kafka.connect.file.FileStreamSourceConnector'
+    FILE_SINK_CONNECTOR = 'org.apache.kafka.connect.file.FileStreamSinkConnector'
+
+    COMMON_GROUP = 'Common'
+    # The value denote whether the config is required or not.
+    COMMON_CONNECTOR_CONFIGS = {
```

Comment:
I think this came up during unit test review too -- this is pretty exhaustive and detailed for a system test. Rather than evaluating high level functionality, this seems like it is evaluating all the minute details. Do we need this level of detail in the system test and is checking, e.g., the specific value of `importance` (rather than, for example, just that the field exists) going to add much to this test?

At a high level, I think we want this test to validate the API fails or succeeds with some basic sets of values. We've gotten away with `==` checks on some things so far because the amount of info was pretty minimal or trivially defined. but I don't know that we want things quite as exacting as they are here -- any minor change to the API, set of fields included, etc will cause this to break. This makes the system tests a lot harder to maintain. (And in fact I think some of the other checks should probably be made a bit more liberal).

I know we need to construct a valid set of configs for each connector, but maybe we should loosen up the assertions here? Maybe split it into fields we *require* to have certain values (which is probably just one or two key ones) and fields we expect to exist (but don't assert specific values for)? Maybe we can find a balance that's more reasonable to maintain given the possibility of extensions to the current set of ConfigDef properties but still validates the API works as expected?